### DISCUSSION TITLE
Create users and gate the project

### DISCUSSION THREAD

**Discussion by cristiandrei1234** [FlowiseAI/Flowise#244] [Ideas]
I thing would be interesting to add the posibility to gate the content and add the posibility to add users to the project, for them to see edit and manage the flows. Or a backend like supabase or firebase.

**Comment by HenryHengZJ** [comment]
We would love to add multi tenant architecture in future!

**Reply by NISHANTSHRIVASTAV** [reply]
Hi @HenryHengZJ,

Any plans where we can have the multi-tenant architecture for Flowise

e.g. Different users should have access to their own chat flows only

Please let us know if it's on the roadmap.

Thanks a lot!

**Comment by hemati** [comment]
Any updates on this?

**Comment by matthias** [comment]
Hi @HenryHengZJ - for now it's SQLite - which just works out of the box.
What's your opinion on Supabase - as it also provides Authenticators, real-time and even a VectorStore - on the downside it's more effort to run it locally. And opinions / experiences here in the community?

**Reply by HenryHengZJ** [reply]
supabase definitely crossed my mind, still researching for other options, as you said it adds another dependency and setup on locally hosted supabase which could be an issue to many non-techies

**Reply by hemati** [reply]
I would have an idea that is a combination of the current version with SQLite and Firebase (just for user authentification). 
We can use the client side authentication of Firebase and on the server side we extend the SQLite databases with another column, namely user_id (which comes from Firebase).

When adding chatflows, api keys, etc to the SQLite DB we then include the user_id.
When querying the chatflows, api keys, etc. from the SQLite we also enter the user_id.

**Reply by paulVu** [reply]
Supabase is great and easy to setup. Please support it

**Comment by huotarih** [comment]
Supabase would be great for this.

**Comment by thiagolealassis** [comment]
+1

**Comment by iaminawe** [comment]
Also adding my support for this, supabase can supply the auth, the managed Postgres (relational and vector) and even the s3 compatible storage. It should definitely be optional as flowise is great as it is but as far as  providing infrastructure to assist with (optionally) making flowise multi tenant via config, I think it would be a game changer
